chromadb/utils/data_loaders.py

```diff
@@ -22,3 +23,120 @@ def _load_image(self, uri: Optional[URI]) -> Optional[Image]:
     def __call__(self, uris: Sequence[Optional[URI]]) -> List[Optional[Image]]:
         with ThreadPoolExecutor(max_workers=self._max_workers) as executor:
             return list(executor.map(self._load_image, uris))
+
+# --------------------------- A lightweight util ---------------------------
+
+# TODO: Placing this here, since data loaders are a common use case for this function.
+#   But it should probably be moved to a more general-purpose module.
+#   See: https://github.com/chroma-core/chroma/issues/1606
+def vectorize(func, iterable=None):
```

Comment:
Here, I'm tempted to add the capability to parallelize the mapping. 
Like this:

```python
from functools import partial
from concurrent.futures import ThreadPoolExecutor

def vectorize(func, iterable=None, *, max_workers : int = 1):
    """Like builtin map, but returns a list, 
    and if iterable is None, returns a partial function that can directly be applied 
    to an iterable.
    This is useful, for instance, for making a data loader from any single-uri loader.
    
    Example:
    >>> vectorize(lambda x: x**2, [1,2,3])
    [1, 4, 9]
    >>> vectorized_square = vectorize(lambda x: x**2)
    >>> vectorized_square([1,2,3])
    [1, 4, 9]
    """
    if iterable is None:
        return partial(vectorize, func, max_workers=max_workers)
    if max_workers == 1:
        return list(map(func, iterable))
    else:
        with ThreadPoolExecutor(max_workers=max_workers) as executor:
            return list(executor.map(func, iterable))
```

It's not the dead-simple vectorize function anymore, but might be more useful in general.

For example, we could use it in all our `DataLoader.__call__` methods.